src/SSTIOGroup.cpp

```diff
@@ -0,0 +1,824 @@
+                               "SST-TF is enabled" } }},
+                // Signal
+                0x01, 0x00
+            },
+        },
+        { "SST::COREPRIORITY_ENABLE",
+            // 0x03 when enabling; 0x01 when disabling
+            { SSTIOGroup::SSTMailboxCommand::CORE_PRIORITY,
+                // Control
+                0x02, 0x100,
+                {{ "ENABLE", { 0x01, 1, 1, M_UNITS_NONE, "SST-CP is enabled" } },
+                 { "DISABLE_RMID_REPORTING", { 0x01, 0, 0, M_UNITS_NONE, "SST RMID reporting is disabled"  } }},
+                // Signal
+                0x02, 0x00
+            },
+        },
+    };
+
+    const std::map<std::string, SSTIOGroup::sst_control_mmio_raw_s> SSTIOGroup::sst_control_mmio_info = {
+        { "SST::COREPRIORITY:0",
+          { GEOPM_DOMAIN_PACKAGE, 0x08,
+            { { "WEIGHT", { 4, 7, 1.0, M_UNITS_NONE, "Proportional priority for core priority level 0"  } },
+              { "FREQUENCY_MIN", { 8, 15, 1e-8, M_UNITS_HERTZ, "Minimum frequency of core priority level 0"  } },
+              { "FREQUENCY_MAX", { 16, 23, 1e-8, M_UNITS_HERTZ, "Maximum frequency of core priority level 0"  } } } } },
+        { "SST::COREPRIORITY:1",
+          { GEOPM_DOMAIN_PACKAGE, 0x0c,
+            { { "WEIGHT", { 4, 7, 1.0, M_UNITS_NONE, "Proportional priority for core priority level 1"  } },
+              { "FREQUENCY_MIN", { 8, 15, 1e-8, M_UNITS_HERTZ, "Minimum frequency of core priority level 1"  } },
+              { "FREQUENCY_MAX", { 16, 23, 1e-8, M_UNITS_HERTZ, "Maximum frequency of core priority level 1"  } } } } },
+        { "SST::COREPRIORITY:2",
+          { GEOPM_DOMAIN_PACKAGE, 0x10,
+            { { "WEIGHT", { 4, 7, 1.0, M_UNITS_NONE, "Proportional priority for core priority level 2"  } },
+              { "FREQUENCY_MIN", { 8, 15, 1e-8, M_UNITS_HERTZ, "Minimum frequency of core priority level 2"  } },
+              { "FREQUENCY_MAX", { 16, 23, 1e-8, M_UNITS_HERTZ, "Maximum frequency of core priority level 2"  } } } } },
+        { "SST::COREPRIORITY:3",
+          { GEOPM_DOMAIN_PACKAGE, 0x14,
+            { { "WEIGHT", { 4, 7, 1.0, M_UNITS_NONE, "Proportional priority for core priority level 3"  } },
+              { "FREQUENCY_MIN", { 8, 15, 1e-8, M_UNITS_HERTZ, "Minimum frequency of core priority level 3"  } },
+              { "FREQUENCY_MAX", { 16, 23, 1e-8, M_UNITS_HERTZ, "Maximum frequency of core priority level 3"  } } } } },
+        { "SST::COREPRIORITY",
+          { GEOPM_DOMAIN_CORE, 0x20, /* offset will be augmented by core index */
+            { { "ASSOCIATION", { 16, 17, 1.0, M_UNITS_NONE, "Assigned core priority level"  } } } } },
+    };
+
+    SSTIOGroup::SSTIOGroup(const PlatformTopo &topo, std::shared_ptr<SSTIO> sstio)
+        : m_topo(topo)
+        , m_sstio(sstio)
+        , m_is_read(false)
+        , m_signal_available()
+        , m_control_available()
+        , m_signal_pushed()
+        , m_control_pushed()
+    {
+        if (m_sstio == nullptr) {
```

Comment:
Nit:  I'd prefer to use the bool operator here, but I'm sure comparison with nullptr is okay.

```
if (!m_sstio) {
...
```
https://en.cppreference.com/w/cpp/memory/shared_ptr/operator_bool

Reply:
The current implementation uses `operator==(const shared_ptr&, nullptr_t)` rather than casting. I'll change it to use bool.

Reply:
Oh if @dianarg were actively reviewing, I'm sure she'd disagree with my recommendation. Given that there is also an == operator specifically for nullptr_t, I'm not sure my recommendation is *clearer to the reader*.